source/fundamentals/data-formats/custom-types.txt

```diff
@@ -1,348 +1,342 @@
-
-Note that the ``transform_bson`` method of the base codec class results in
-these values being decoded as ``Decimal`` (and not ``DecimalInt``).
-
-
-.. _decoding-binary-types:
-
-Decoding ``~bson.binary.Binary`` Types
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-The decoding treatment of ``~bson.binary.Binary`` types having
-``subtype = 0`` by the ``bson`` module varies slightly depending on the
-version of the Python runtime in use. This must be taken into account while
-writing a ``TypeDecoder`` that modifies how this datatype is decoded.
-
-On Python 3.x, ``~bson.binary.Binary`` data (``subtype = 0``) is decoded
-as a ``bytes`` instance:
-
-.. code-block:: pycon
+   class DecimalIntCodec(DecimalCodec):
+       @property
+       def python_type(self):
+           # The Python type acted upon by this type codec
+           return DecimalInt
 
-    >>> # On Python 3.x.
-    >>> from bson.binary import Binary
-    >>> newcoll = db.get_collection("new")
-    >>> newcoll.insert_one({"_id": 1, "data": Binary(b"123", subtype=0)})
-    >>> doc = newcoll.find_one()
-    >>> type(doc["data"])
-    bytes
+You can then add the sublcass's type codec to the type registry and encode instances
+of the custom type:
 
+.. io-code-block::
+   :copyable: true
 
-On Python 2.7.x, the same data is decoded as a ``~bson.binary.Binary``
-instance:
+   .. input::
+      :language: python
 
-.. code-block:: pycon
+      import pprint
+      from bson.codec_options import CodecOptions
 
-    >>> # On Python 2.7.x
-    >>> newcoll = db.get_collection("new")
-    >>> doc = newcoll.find_one()
-    >>> type(doc["data"])
-    bson.binary.Binary
+      decimal_int_codec = DecimalIntCodec()
+      type_registry = TypeRegistry([decimal_codec, decimal_int_codec])
+      codec_options = CodecOptions(type_registry=type_registry)
 
+      collection = db.get_collection("test", codec_options=codec_options)
+      collection.drop()
+      collection.insert_one({"num": DecimalInt("45.321")})
+  
+      my_doc = collection.find_one()
+      pprint.pprint(my_doc)
 
-As a consequence of this disparity, users must set the ``bson_type`` attribute
-on their ``~bson.codec_options.TypeDecoder`` classes differently,
-depending on the python version in use.
+   .. output::
+      :language: shell
 
+      {'_id': ObjectId('...'), 'num': Decimal('45.321')}
 
 .. note::
+  
+   The ``transform_bson()`` method of the ``DecimalCodec`` class results in
+   these values being decoded as ``Decimal``, not ``DecimalInt``.
 
-  For codebases requiring compatibility with both Python 2 and 3, type
-  decoders will have to be registered for both possible ``bson_type`` values.
+Define a Fallback Encoder
+-------------------------
 
+You can also register a **fallback encoder**, a callable to encode types not recognized
+by BSON and for which no type codec has been registered.
+Like the ``transform_python()`` method, the fallback encoder accepts an unencodable
```

Comment:
S: I don't think this is particularly relevant
```suggestion
The fallback encoder accepts an unencodable
```

Reply:
i'll buy that